build.gradle

```diff
@@ -1,12 +1,58 @@
+buildscript {
+    /**
+     * Returns sinceBuild for a given IC- version.
+     *
+     * Supported cases:
+     * (1) Year and Version -- (a) IC-2017.3, (b) 2017.2.4, (c) 2018.1
+     * (2) Build ID -- (a) IU-163.7890, (b) 171.1234
+     * (3) Named Snapshots -- (a) LATEST-EAP-SNAPSHOT, (b) LATEST-TRUNK-SNAPSHOT
+     *
+     * Taken and adapted from https://github.com/rhdunn/xquery-intellij-plugin/blob/master/build.gradle#L1-L47
+     */
+    ext.getSinceIdeaBuild = { version ->
```

Comment:
That seems too complex to me if you need to convert ideaVersion to a 3 digit why don't you use substring ?

Reply:
replaced by a simple transformation ideaVersion->sinceIdeaBuild